### PR TITLE
test(test runner): calling test.step() from fixture

### PR DESCRIPTION
**Pull request by dgozman**

### CODE REVIEW

File: tests/playwright-test/test-step.spec.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    'reporter.ts': expectReporterJS,
+    'playwright.config.ts': `
+      module.exports = {
+        reporter: './reporter',
+      };
+    `,
+    'a.test.ts': `
+      const test = pwt.test.extend({
+        foo: async ({}, use) => {
+          await pwt.test.step('setup foo', () => {});
```
Comment (JoelEinbinder) [inline tests/playwright-test/test-step.spec.ts:154]: Does this work if the test object was extended again? How about if this was using the current `test` variable?

File: tests/playwright-test/test-step.spec.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+          await pwt.test.step('teardown foo', () => {});
+        },
+      });
+      test('pass', async ({ foo }) => {
+        await test.step('test step', async () => {
+          await foo();
+        });
+      });
+    `
+  }, { reporter: '', workers: 1 });
```
Comment (JoelEinbinder) [inline tests/playwright-test/test-step.spec.ts:167]: what does `reporter: ''` do?

File: tests/playwright-test/reporter.spec.ts [hunk 3]
```diff
@@ -337,41 +337,6 @@ test('should report api step failure', async ({ runInlineTest }) => {
   ]);
 });
 
-test('should report test.step', async ({ runInlineTest }) => {
```
Comment (JoelEinbinder) [inline tests/playwright-test/reporter.spec.ts:340]: Where did this test go?

**Review by JoelEinbinder — APPROVED** [review]

File: tests/playwright-test/reporter.spec.ts (hunk 3 above)
Comment (dgozman) [reply tests/playwright-test/reporter.spec.ts:340]: It's superseded by other tests.

**Pull request merged**